### PR TITLE
[stable/kubernetes-dashboard] set ingress Secure-Backends as default value

### LINKED ISSUE #4204: [stable/kubernetes-dashboard] TLS handshake error: first record does not look like a TLS handshake
**Is this a request for help?**: Yes, as the chart is not working as expected.

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

**Version of Helm and Kubernetes**: helm version is 2.8.1 and kubernetes version is 1.8.6

**Which chart**: kubernetes-dashboard

**What happened**:
I have installed kubernetes-dashboard using helm. The command used was
```
helm install --name kubernetes-dashboard stable/kubernetes-dashboard --namespace kube-system --set rbac.create=false --set ingress.enabled=true
```
But then when am accessing it using the ingress hostname its giving and SSL related error. As per pod logs
```
2018/03/16 14:58:41 http: TLS handshake error from 100.96.3.7:56916: tls: first record does not look like a TLS handshake
```
The ingress looks like
```
metadata:
  annotations:
    kubernetes.io/tls-acme: "true"
    nginx.ingress.kubernetes.io/auth-signin: https://$host/oauth2/sign_in
    nginx.ingress.kubernetes.io/auth-url: https://$host/oauth2/auth
  creationTimestamp: 2018-03-16T14:16:48Z
  generation: 3
  labels:
    app: kubernetes-dashboard
    chart: kubernetes-dashboard-0.6.1
    heritage: Tiller
    release: kubernetes-dashboard
  name: kubernetes-dashboard
  namespace: kube-system
  resourceVersion: "4072579"
  selfLink: /apis/extensions/v1beta1/namespaces/kube-system/ingresses/kubernetes-dashboard
  uid: aa54f4f0-2924-11e8-aedc-0a717f21e61e
spec:
  rules:
  - host: dashboard.atkubernetes-lab.com
    http:
      paths:
      - backend:
          serviceName: kubernetes-dashboard
          servicePort: 443
        path: /
  tls:
  - hosts:
    - dashboard.atkubernetes-lab.com
    secretName: dashboard-tls
```
I am able to access the dashboard via API and token, but just the via the ingress resource it's not and ending up in error as I mentioned above.

**What you expected to happen**:
I am expecting to have the dashboard rendering properly when access via https://dashboard.atkubernetes-lab.com.

**How to reproduce it** (as minimally and precisely as possible):

**Anything else we need to know**:
It could be a config issue on my side, but any insights would be of great help. I am using kubernetes-dashboard on few of my other clusters, which are installed using helm. But they are not on the latest version, in the sense its using the image before 443 port was enforced.

Thanks

### PR DESCRIPTION
**Pull request by MnrGreg**
Sets Secure-Backends = "true" by default to fix #4204 without requiring values changes.

Note it is currently not possible to set this configuration via the CLI. Helm converts "true" to boolean. With future helm 2.9.0 release one will be able to force this configuration as string using the "--set-string" option. For example:
`helm install stable/kubernetes-dashboard --set-string ingress.annotations."nginx\.ingress\.kubernetes\.io/secure-backends"="true"`

This PR sets the 'secure-backends' as a default value within the charts values.yaml file. At the moment the README states that "kubernetes.io/ingress.class: nginx" is already a default setting. Therefor adding the secure-backends should not be too prescriptive either. Ingress environments still require that ingress.enabled=false be overridden to true in values.yaml.

Maintainers
@kfox1111

### CODE REVIEW

**Comment by kfox1111** [comment]
I haven't heard that there was a way yet to load the backends ca into the ingress-controller smoothly yet. So setting the secure backend flag only works if you somehow have out of band done so. I'd recommend defaulting it to false until nginx-ingress has a path forward for ingress specified ca's.

Uncommenting out the ingress class setting is also a breaking change I believe. Folks may not be using nginx, and after this change it will switch it to nginx unless they re-specify the option.

File: stable/kubernetes-dashboard/README.md [hunk 1]
```diff
@@ -52,7 +52,7 @@ The following table lists the configurable parameters of the kubernetes-dashboar
 | `tolerations`             | List of node taints to tolerate (requires Kubernetes >= 1.6)                                                                | `[]`                                                                     |
 | `service.externalPort`    | Dashboard external port                                                                                                     | 443                                                                      |
 | `service.internalPort`    | Dashboard internal port                                                                                                     | 443                                                                      |
-| `ingress.annotations`     | Specify ingress class                                                                                                       | `kubernetes.io/ingress.class: nginx`                                     |
+| `ingress.annotations`     | Specify ingress class                                                                                                       | `kubernetes.io/ingress.class: nginx`, `nginx.ingress.kubernetes.io/secure-backends: "true"`   |
```
Comment (unguiculus) [inline stable/kubernetes-dashboard/README.md]: Remove the defaults as there are none.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/kubernetes-dashboard/README.md (hunk 1 above)
Comment (MnrGreg) [reply stable/kubernetes-dashboard/README.md]: Removed default from README

**Review by unguiculus — APPROVED** [review]

**Comment by kfox1111** [comment]
No one ever responded to my question. :( without loading a cert somehow into the ingress container, it will always fail. Please revert asap.

**Comment by kfox1111** [comment]
oh. nm. just saw it was commented out.

**Pull request merged**